pkg/scheduler/factory/cache_comparer.go

```diff
@@ -95,37 +96,37 @@ func (c compareStrategy) CompareNodes(nodes []*v1.Node, nodeinfos map[string]*sc
 	return compareStrings(actual, cached)
 }
 
-func (c compareStrategy) ComparePods(pods, waitingPods []*v1.Pod, nodeinfos map[string]*schedulercache.NodeInfo) (missed, redundant []string) {
-	actual := []string{}
+func (c compareStrategy) ComparePods(pods, waitingPods []*v1.Pod, nodeinfos map[string]*schedulercache.NodeInfo) (missed, redundant []types.UID) {
```

Comment:
You didn't need to change almost anything in this file. You can keep everything as it was and just use pdb.UID instead of pdb.Name at line 120.